services/blockchain-indexer/shared/dataService/business/interoperability/blockchainApps.js

```diff
@@ -71,11 +75,30 @@ const getBlockchainApps = async (params) => {
 
 	const total = await blockchainAppsTable.count(params);
 
-	blockchainAppsInfo.data = await blockchainAppsTable.find(
+	const resultSet = await blockchainAppsTable.find(
 		{ ...params, limit: params.limit || total },
 		Object.getOwnPropertyNames(blockchainAppsTableSchema.schema),
 	);
 
+	const { chainID } = (await getNetworkStatus()).data;
+
+	blockchainAppsInfo.data = await BluebirdPromise.map(
+		resultSet,
+		async blockchainAppInfo => {
+			const { escrowedAmounts } = await requestConnector('getEscrowedAmounts');
```

Comment:
since it's not dependant on the `blockchainAppInfo` we can fetch before the loop, so that we do not have to invoke multiple times.